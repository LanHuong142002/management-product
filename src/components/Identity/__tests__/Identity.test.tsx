import { cleanup, render, screen } from '@testing-library/react';
import { Identity } from '@components';

describe('Testing identity component', () => {
  afterEach(() => {
    cleanup();
  });

  const defaultProps = {
    url: '',
    alt: 'vendôme louis',
    text: 'vendôme louis',
  };

  it('Should render Identity', () => {
    const { container } = render(<Identity {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  it('Should render Identity with circle shape', () => {
    render(<Identity {...defaultProps} isCircle={true} />);

    const identity = screen.getByTestId('identity-wrapper');

    expect(identity).toBeInTheDocument();
  });

  it('Should render Identity with default shape', () => {
    render(<Identity {...defaultProps} isCircle={false} />);

    const identity = screen.getByTestId('identity-wrapper');

    expect(identity).toBeInTheDocument();
  });
});
