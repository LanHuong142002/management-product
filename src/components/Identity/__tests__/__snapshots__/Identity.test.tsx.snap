// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing identity component Should render Identity 1`] = `
<div>
  <div
    class="identity-wrapper"
    data-testid="identity-wrapper"
  >
    <figure
      class="image-wrapper image-size-lg "
    >
      <img
        alt="vendôme louis"
        class="image "
        src=""
      />
    </figure>
    <span>
      vendôme louis
    </span>
  </div>
</div>
`;
